### DISCUSSION TITLE
texlive problems

### DISCUSSION THREAD

**Discussion by revelator** [msys2/msys2-texlive#12] [Q&A] [answered]
hmm some problems im seing while building some of the extra packages for texlive.

`fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `luatex -ini   -jobname=dviluatex -progname=dviluatex dviluatex.ini <nul' return status: 1
fmtutil [ERROR]: returning error due to option --strict
fmtutil [ERROR]: running `pdftex -ini   -jobname=etex -progname=etex -translate-file=cp227.tcx *etex.ini <nul' return status: 1`

is this due for concern ?

**Comment by naveen521kk** [comment]
Did this happen when running hooks? Also, is that all the problem?

**Comment by revelator** [comment]
happened while building some of the texlive packages, not the only problem no but the error is a bit long so maybe better if i put it on pastebin.

**Reply by naveen521kk** [reply]
I have seen the error once happened because it missed some features, which is fine to be ignored(unless those features are enabled which building texlive-bin).

**Comment by revelator** [comment]
another problem was spawning cmd.exe if that helps, i checked paths and all seem in order so im not sure if it has to do with some path muckup. I got it to shut up about not being able to spawn cmd.exe by symlinking it to /mingw*/bin/ the weird thing is i had no problem executing cmd.exe from bash even before i used this hack.

**Comment by revelator** [comment]
https://pastebin.com/AjV75dZX

Hmm seems like it is caused by my locale which resolves to da_DK.UTF-8 if i change it to en_GB it seems to work.

**Reply by revelator** [reply]
Well it got a bit further but still getting spawn cmd.exe error and it still bails when running fmtutil :S
it does build the package though but im not sure it works correctly because of these errors.

**Reply by naveen521kk** [reply]
Looks like the error are a bit weird. I couldn't reproduce it. 

You can get the full log output by changing https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-texlive-core/texlive-fmtutil.script.in#L55, see if that helps.

**Reply by revelator** [reply]
ill try the script change hang on.

**Reply by revelator** [reply]
https://pastebin.com/6A52hN6E

looks much the same though even with the new script.

EDIT: no the spawn cmd.exe errors have gone atleast.

**Reply by naveen521kk** [reply]
Did you change that line to 
`@MINGW_PREFIX@/bin/fmtutil-sys --all --cnffile $FMTUTIL`
removing that `/dev/null` redirect? AFAIK, the output should be still large (see [attachment](https://tug.org/pipermail/tldistro/2021q2/000421.html))

**Reply by revelator** [reply]
Ah misunderstood what you wanted me to do, removed the /dev/null from the script now.
Doing another compile in a sec.

**Reply by revelator** [reply]
welp!! to big to copy from the console so ill do a  2>&1 | tee -a install.log and upload it here zipped if thats ok ?.

**Reply by revelator** [reply]
[install32.zip](https://github.com/msys2/MINGW-packages/files/6502149/install32.zip)

here you go i noiticed some errors creating pdf output but not sure what gives.

**Reply by naveen521kk** [reply]
can you send all the files matching `c:/Msys2/mingw32/etc/texmf/tex/generic/config/language*` (put it in a zip file and upload).

it says something like that some file was wrongly formed.
```
(c:/Msys2/mingw32/etc/texmf/tex/generic/config/language.def...sys2/mingw32/etc/texm
f/tex/generic/config/language.dat.lua:54: unexpected symbol near ','
stack traceback:
	[C]: in function 'dofile'
	.../share/texmf-dist/tex/luatex/hyph-utf8/luatex-hyphen.lua:30: in main chunk
	[C]: in function 'require'
	[\directlua]:1: in main chunk.
```

**Reply by revelator** [reply]
sure :) 
[language-config.zip](https://github.com/msys2/MINGW-packages/files/6502464/language-config.zip)

**Reply by revelator** [reply]
whoops wrong mingw installation the zipped files are from an experimental mingw configuration heres the right ones->
[language-config-mingw32.zip](https://github.com/msys2/MINGW-packages/files/6502507/language-config-mingw32.zip)

**Comment by revelator** [comment]
hmm looks like the line in question is empty -> lefthyphenmin = , should be lefthyphenmin = 2, and righthyphenmin = , should be righthyphenmin = 3, according to the same file from an official texlive installation.

**Reply by naveen521kk** [reply]
Can you send `c:/Msys2/mingw32/var/lib/texmf/msys2/installedpkgs/*` also?

**Reply by naveen521kk** [reply]
what you tell is correct. I think there is some bug while parsing possibly in https://github.com/msys2/msys2-texlive. Maybe, I will try to fix it. 
I could see the file exists here also, https://github.com/msys2/msys2-texlive/releases/download/20210506/texlive-lang-arabic-extra-files.tar.xz, so it should be a bug there.
```lua
-- test-- from hyphen-arabic:
['arabic'] = {
    loader = 'zerohyph.tex',
    lefthyphenmin = ,
    righthyphenmin = ,
    synonyms = {  },
    patterns = '',
    hyphenation = '',
},
-- from hyphen-farsi:
['farsi'] = {
    loader = 'zerohyph.tex',
    lefthyphenmin = ,
    righthyphenmin = ,
    synonyms = { 'persian' },
    patterns = '',
    hyphenation = '',
},
```
Can you make an issue in that Repo?

**Reply by naveen521kk** [reply]
so looking at the `texlive.tlpdb` file from https://github.com/msys2/msys2-texlive/releases/download/20210506/texlive-lang-arabic-extra-files.tar.xz, which was parsed to build this, I could find this line
```
execute AddHyphen 	name=arabic 	lefthyphenmin= 	righthyphenmin= 	file=zerohyph.tex 	file_patterns=
```
which looked weird (well my parser's worked correctly) but it should be a bug in that `texlive.tlpdb` at the time of packaging. Maybe I should report upstream? (@norbusan)

**Reply by naveen521kk** [reply]
Sadly, this still exists the latest version from https://mirror.unpad.ac.id/ctan/systems/texlive/tlnet/tlpkg/texlive.tlpdb. So, it's something people from the texlive team should fix. Maybe I will write an email.

**Reply by revelator** [reply]
Sounds like a good idea :) zip seems to have trouble uploading atm ? maybe it is to big.

**Reply by revelator** [reply]
[installedpkgs.zip](https://github.com/msys2/msys2-texlive/files/6502823/installedpkgs.zip)

**Reply by revelator** [reply]
there we go not sure what happened before ...

**Reply by naveen521kk** [reply]
I have sent https://tug.org/pipermail/tex-live/2021-May/046962.html. Let's see what happens...

After they have fixed I will update it :).

**Reply by revelator** [reply]
cool :)

**Reply by naveen521kk** [reply]
Ok, got a reply from [upstream](https://tug.org/pipermail/tex-live/2021-May/046963.html) and accepted it was a bug. On the other hand, I was advised to set default values while parsing it https://tug.org/pipermail/tex-live/2021-May/046964.html, so that it doesn't create weird things like this.
So, I am going to have some default values while parsing it and roll a new release.

**Comment by naveen521kk** [comment, marked as answer]
@revelator you can try the pkgbuild's from https://github.com/msys2/MINGW-packages/pull/8715

**Reply by revelator** [reply]
hmm im getting build errors regarding freetype with the new packages.
seems to be the deprecated macro we been having problems with in other packages -> #define FT_CID_H  <freetype/ftcid.h> this file does not exist anymore but seems the devs forgot to remove it from the build :S

it is located in /mingw*/include/freetype2/freetype/config/ftheader.h

**Reply by revelator** [reply]
well even uncommenting it is not enough seems texlive includes it by default.

**Reply by revelator** [reply]
got around it by copying of the missing header from source since it is still there but newer installed.

**Reply by naveen521kk** [reply]
You don't need to build `texlive-bin` though. 
And, I think the cause for that is because we were using the master branch of FreeType because it contains meson support and could have been breaking https://github.com/msys2/MINGW-packages/pull/8535 cc @lazka

**Reply by revelator** [reply]
oh ok well just for completeness :) i also linked the missing header patch for the meson build.
one other thing the backup section in texlive-core does not work with forward slashes, this however fixes it.
`backup=("${MINGW_PREFIX:1}\\etc\\texmf\\web2c\\texmf.cnf"
        "${MINGW_PREFIX:1}\\etc\\texmf\\dvipdfmx\\dvipdfmx.cfg"
        "${MINGW_PREFIX:1}\\etc\\texmf\\dvips\\config\\config.ps"
        "${MINGW_PREFIX:1}\\etc\\texmf\\tex\\generic\\config\\language.dat"
        "${MINGW_PREFIX:1}\\etc\\texmf\\tex\\generic\\config\\language.def"
        "${MINGW_PREFIX:1}\\etc\\texmf\\tex\\generic\\tex-ini-files\\pdftexconfig.tex"
        "${MINGW_PREFIX:1}\\etc\\texmf\\web2c\\fmtutil.cnf"
        "${MINGW_PREFIX:1}\\etc\\texmf\\web2c\\mktex.cnf"
        "${MINGW_PREFIX:1}\\etc\\texmf\\xdvi\\XDvi")`

i noticed that the backups were newer created and if doing an updpkgsums pacman complained about it also, the above fixes that.

**Reply by revelator** [reply]
another one in texlive-extra-utils

`backup=(
    "${MINGW_PREFIX:1}\\etc\\texmf\\chktex\\chktexrc"
)`

**Reply by naveen521kk** [reply]
> one other thing the backup section in texlive-core does not work with forward slashes

Really? I don't much about this. Maybe can you make a PR?

**Reply by revelator** [reply]
ill see if i can get around to it :)

**Comment by revelator** [comment]
[missing-header.zip](https://github.com/msys2/msys2-texlive/files/6520092/missing-header.zip)
 
patch for the missing freetype header
